apps/sms/test/marionette/lib/messages.js

```diff
@@ -237,16 +237,38 @@
               break;
             }
           }
-
-          client.helper.waitForElementToDisappear(menuElement);
         },
 
         addRecipient: function(number) {
+          // "recipientschange" event is fired once for currently edited
+          // recipient (via "sendKeys" below) and four times during recipient
+          // assimilation. See bug 1109801.
+          var numberOfRecipientsChangeEvents = 5;
+
+          // Since recipient.js re-renders recipients all the time (when new
+          // recipient is added or old is removed) and it can happen several
+          // times during single "add" or "remove" operation we should
+          // wait until Recipients View is in a final state. The problem here is
+          // that between "findElement" and "displayed" calls element can
+          // actually be removed from DOM and re-created again that will lead to
+          // "stale element" exception.
+          client.executeScript(function(numberOfEvents) {
```

Comment:
@steveck-chung I know that it looks quite scary :) But can't think of another hook we can use other than timeout. Maybe you have better ideas?